update track selection page, use only min price for course modes

zubair-arbi opened the pull request:
ECOM-1228

WIP
1. Use only min price in track selection pages with amount shown in submit/payment buttons for all course modes [honor/audit, verified, professional, no-id-professional]
2. Remove 'suggested_prices' field from 'CourseMode' admin

@awais786 @aamir-khan @wedaly

zubair-arbi commented:
test this please

zubair-arbi commented:
@wedaly @AlasdairSwan Please review.
Sandbox: http://zubair-arbi.m.sandbox.edx.org

wedaly commented on common/djangoapps/course_modes/models.py:
```diff
@@ -553,8 +553,8 @@ def to_tuple(self):
-        return u"{} : {}, min={}, prices={}".format(
-            self.course_id.to_deprecated_string(), self.mode_slug, self.min_price, self.suggested_prices
+        return u"{} : {}, min={}".format(
+            self.course_id.to_deprecated_string(), self.mode_slug, self.min_price   # pylint: disable=no-member
```
`to_deprecated_string()` is no longer necessary -- you can just use `self.course_id` here and it will be automatically typecast to `unicode`

wedaly commented on common/templates/course_modes/choose.html:
```diff
@@ -62,13 +62,9 @@ <h3 class="title">${_("Sorry, there was an error when trying to enroll you")}</h
-                    % if not upgrade:
-                        <h3 class="title">${_("Now choose your course track:")}</h3>
-                    % endif
+                    <%include file="/verify_student/_verification_header.html" args="course_name=course_name" />
```
Why move this header?

wedaly commented on common/templates/course_modes/choose.html:
```diff
@@ -77,39 +73,31 @@ <h3 class="title">${_("Now choose your course track:")}</h3>
-                                            <div class="copy">
-                                                <p><i class="msg-icon icon fa fa-exclamation-triangle"></i> ${error}</p>
+                                    <div class="copy">
+                                        <p>${_("Highlight you new knowledge and skills with a verified certificate. Use this valuable credential to improve your job prospects and advance your career, or highlight your certificate in  school applications.")}</p>
```
Small typo here: two spaces instead of one for "in  school"

wedaly commented on lms/templates/verify_student/_verification_header.html:
```diff
@@ -2,71 +2,28 @@
+            )}
         % endif
-    </h2>
+    </h3>
```
Why are these changes necessary?

wedaly commented on lms/djangoapps/verify_student/views.py:
```diff
@@ -376,6 +377,19 @@ def get(
+                all_course_modes
+            )
+            if verify_status_by_course:
+                verification_good_until = verify_status_by_course[course_key].get('verification_good_until') or ''
```
I think this is more complicated than what you need.  Why not just retrieve the most recent software secure photo verification for the course with status either "submitted" or "approved", then use the date from that?

We're already making a similar query in `_check_already_verified` -- we should need only one database query to check whether (a) the user has a submitted or approved verification attempt, and (b) the expiration date of that attempt.

wedaly commented:
I had a few comments that I deleted after reading the discussion in the JIRA ticket.

One last point: @clrux recently made some a11y changes to this flow (which I suspect are the cause of the merge conflicts).  Could you please make sure that his a11y changes aren't accidentally reverted?

zubair-arbi commented:
@wedaly I have addressed your comments. Rebsed onto master and squashed commits.
Split `user_has_valid_or_pending` into two methods [`user_has_valid_or_pending`, `verification_valid_or_pending`] and created `_verification_valid_until` method to get expiration date of valid verification regarding a user using `verification_valid_or_pending` method.
@clrux please review

clrux commented:
@zubair-arbi Were all the line deletions intentional?

zubair-arbi commented:
@clrux yes. This is because we will now show min price on all course modes. Please check screen shots on ticket ECOM-1228 for further details.

clrux commented:
So far as I can tell here, this is ok :+1:

wedaly commented on lms/djangoapps/verify_student/views.py:
```diff
@@ -579,6 +584,28 @@ def _requirements(self, display_steps, is_active):
+
+        Returns:
+            datetime object or datetime object in string format if
+            'date_format' argument is provided
```
Do we need this flexibility?  It seems like in our case, we always want the formatted string.  I'd like to avoid returning different types from the same method if possible.

wedaly commented:
One small style comment, otherwise :+1:
